packages/jellyfish-api-core/__tests__/category/masternode/createMasternode.test.ts

```diff
@@ -0,0 +1,76 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { MasternodeState } from '../../../src/category/masternode'
+
+describe('Masternode', () => {
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForReady()
+    await container.waitForWalletCoinbaseMaturity()
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  it('should create a masternode transaction', async () => {
+    const masternodesBefore = await client.masternode.listMasternodes()
+    const masternodesLengthBefore = Object.keys(masternodesBefore).length
+
+    const address = await client.wallet.getNewAddress()
+    const masternodeTransaction = await client.masternode.createMasternode(address)
+
+    await container.generate(1)
+
+    const masternodesAfter = await client.masternode.listMasternodes()
```

Comment:
```suggestion
    const masternodesAfter = (await client.masternode.listMasternodes()).filter(mn => mn.ownerAuthAddress === address)
```

Reply:
I understand why you made that suggestion, but `client.masternode.listMasternodes()` returns an object not an array.

Reply:
@siradji my bad, overlooked that.

But possible apply the filter at `Object.values(masternodesAfter).filter(mn => mn.ownerAuthAddress === address)` right. Then can remove the if at L37 and 100% certain that the node is actually newly created.